src/themes/cornerstone/RegisterSuccess.js

```diff
@@ -0,0 +1,66 @@
+import { hideElementsByClassName } from '../../modules/hideElementByClassName';
+import { popTabToRoot } from '../../modules/app_commands/popTabToRoot';
+import { openPage } from '../../modules/app_commands/openPage';
+import { showTab } from '../../modules/app_commands/showTab';
+import { sendAppCommands } from '../../modules/sendAppCommands';
+
+/**
+ * Makes register success page escape proof
+ */
+export class RegisterSuccess {
+  /**
+   * @param {ShopgateAppCodeExecutor} shopgateAppCodeExecutor to execute Shopgate app related code
+   */
+  constructor(shopgateAppCodeExecutor) {
+    this.shopgateAppCodeExecutor = shopgateAppCodeExecutor;
+  }
+
+  /**
+   * Makes register success page escape proof
+   */
+  execute = () => {
+    this.hideHeader();
+    this.hideFooter();
+    this.hideButton();
+    this.closeInAppBrowser();
+  };
+
+  /**
+   * Will close the In-App-Browser and redirect the user to the register/login page.
+   * In case the user is logged in already the cart is shown
+   * @private
+   */
+  closeInAppBrowser() {
+    this.shopgateAppCodeExecutor.execute(() => {
+      sendAppCommands([
```

Comment:
why do we switch here to the legacy system? 
why do we have to redirect the user to the login page? i thought the user should be already logged in
why not using the `closeInAppBrowser` broadcast event?

Reply:
changed the behavior to send the closeInAppBrowser event